pkg/utils/utils.go

```diff
@@ -319,14 +320,22 @@ func StringSliceEqual(s1, s2 []string) bool {
 		return false
 	}
 
-	for _, s := range s1 {
-		if !StringInSlice(s2, s) {
-			return false
-		}
+	// mapKeys to remember keys that exist in s1
+	mapKeys := map[string]int{}
+
+	// first list all items in s1
+	for _, v := range s1 {
+		mapKeys[v]++
+	}
+
+	// second list all items in s2
+	for _, v := range s2 {
+		mapKeys[v]--
```

Comment:
embedded check here for more efficient 
```go
mapKeys[v]--
if mapKeys[v] < 0 {
return false
}
```
This check covers two situations
1. if the key is not exists, mapKey[v] == -1
2. if the key is redundant, mapKey[v] < 0

Reply:
good advice

Reply:
you are so talented. @zhuangqh 👍 🦁 🐸 🐼